### PR TITLE
SOC-1857 Cache results from getTopWikis query

### PR DESCRIPTION
**Pull request by garthwebb**
Implemented caching of often run query.  Note that this builds off cleanup pull request #9430.

https://wikia-inc.atlassian.net/browse/SOC-1857

### CODE REVIEW

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php [hunk 1]
```diff
@@ -33,67 +39,71 @@ public function __construct( User $user ) {
...
+
+		$wikis[$wikiId] = [
+			'id' => $wikiId,
+			'wikiName' => $wikiName,
+			'wikiUrl' => $wikiUrl,
+			'edits' => $editCount
+		];
+	}
+
+	private function getTopWikisQuery( $limit = null ) {
```
Comment (kvas-damian) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php]: what do you think about `$limit = self::MAX_FAV_WIKIS`?

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php [hunk 2]
```diff
@@ -105,11 +115,18 @@ private function getTopWikisFromDb( $limit = null ) {
...
 	public function getTopWikis( $refreshHidden = false ) {
 
-		if ( $refreshHidden === true ) {
+		if ( $refreshHidden ) {
 			$this->clearHiddenTopWikis();
 		}
 
-		$wikis = array_merge( $this->getTopWikisFromDb(), $this->getEditsWikis() );
+		$memKey = $this->getTopWikisMemKey();
+		$savedWikis = $this->wg->Memc->get( $memKey );
```
Comment (kvas-damian) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php]: What do you think about using `WikiaDataAccess::cache()` here?

Comment (kvas-damian) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php]: I think we could use `SharedMemcKey` for better performance.

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php (hunk 1 above)
Comment (garthwebb) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php]: Sounds better

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php (hunk 2 above)
Comment (garthwebb) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php]: The `getTopWikisMemKey()` method uses `wfSharedMemKey()`. Is that what you mean?

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php [hunk 3]
```diff
@@ -33,67 +39,67 @@ public function __construct( User $user ) {
...
-		if ( is_null( $limit ) ) {
-			$limit = self::MAX_FAV_WIKIS;
-		}
+	private function getTopWikisFromDb( $limit = self::MAX_FAV_WIKIS ) {
+		global $wgCityId, $wgSpecialsDB;
 
 		if ( empty( $wgCityId ) ) {
 			// staff/internal does not have stats db
 			$wikis = [];
-		} else if ( $wgDevelEnvironment ) {
```
Comment (jsutterfield) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php:52]: Do we not have to worry about this `$wgDevelEnvironment` anymore? Looks like it didn't make it into the new version.

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php (hunk 2 above)
Comment (kvas-damian) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php]: Right, I misread the code ;)

**Comment by kvas-damian** [comment]
LGTM

File: extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php (hunk 3 above)
Comment (garthwebb) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php:52]: They were using $wgDevelEnvironment as a way to cook the data in dev for what was essentially guided unit tests.  The `getTestData` method returned a set of preselected wikis, one of which was one that should be filtered.  This really belongs in a unit test somehwere.  Also, there wasn't any "strange behavior" that I could tell, so I thought it was best to get rid of this fake code path.

Comment (jsutterfield) [inline extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php:52]: Okay. Makes sense.

**Comment by jsutterfield** [comment]
LGTM

**Pull request merged**